client/jira.go

```diff
@@ -75,8 +76,28 @@ type (
 	AssignRequest struct {
 		Name string `json:"name,omitempty"`
 	}
+
+	TransitionsResult struct {
+		Transitions []Transition `json:"transitions"`
+	}
+
+	Transition struct {
+		Name string `json:"name"`
+	}
 )
 
+func GetTransitions(issueId string) (TransitionsResult, error) {
+	var results TransitionsResult
+	path := fmt.Sprintf("/rest/api/2/issue/%s/transitions", issueId)
+	request, err := constructGetRequest(path)
+	if err != nil {
+		return results, err
+	}
+	statusCode, err := SendRequest(request, &results)
```

Comment:
I'd like to see some error handling here; the request may fail for any number of reasons, I'd return the err and a nil `TransitionsResult`.